### ISSUE TITLE
Return type self in arrow function gets prefixed

### ISSUE THREAD

**Issue by ondrejmirtes** [humbug/php-scoper#542]
## Bug report

I'm experiencing this with Box version `3.13.0@275b091` because I don't know how to try the latest of everything.

This code:

```php
return array_combine(
            $traitNames,
            array_map(static fn (BetterReflectionClass $trait): self => new self($trait), $traits),
        );
```

gets turned into:

```php
return \array_combine(
            $traitNames
             \array_map(static fn(\PHPStan\BetterReflection\Reflection\ReflectionClass $trait): \_PHPStan_2130649\self => new self($trait), $traits)
        );
```

Expected behaviour: `self` typehint should not be prefixed.

**Comment by leoloso** [comment]
I'm experiencing the same issue, but with `self` as a nullable typed property:

```php
namespace Symfony\Component\DependencyInjection\Compiler;

class AutowirePass extends AbstractRecursivePass
{
    // ...
    private ?self $typesClone = null;
    // ...
}
```

**Comment by theofidry** [comment]
Fixed in #542 and should be part of 0.16
